File: golabs/gocloud/godata/go-gumball-riak/gumball-riak/src/github.com/basho/taste-of-riak/csharp/Ch03-Msgy-Schema/Repositories/UserRepository.cs
namespace Ch03_Msgy_Schema.Repositories
{
    using Models;
    using RiakClient;

    public class UserRepository : Repository<User>
    {
        const string BUCKET_NAME = "Users";

        public UserRepository(IRiakClient client)
            : base(client)
        {
        }

        protected override string BucketName
        {
            get { return BUCKET_NAME; }
        }
    }
}

File: golabs/gocloud/godata/go-gumball-riak/gumball-riak/src/github.com/basho/taste-of-riak/java/Ch03-Msgy-Schema/src/com/basho/msgy/Repositories/UserRepository.java
package com.basho.msgy.Repositories;

import com.basho.msgy.Models.User;
import com.basho.riak.client.IRiakClient;
import com.basho.riak.client.RiakRetryFailedException;
import com.basho.riak.client.bucket.Bucket;

public class UserRepository {
    static final String BUCKET_NAME = "Users";
    protected IRiakClient client;

    public UserRepository(IRiakClient client) {
        this.client = client;
    }

    public void save(User user) throws RiakRetryFailedException {
        Bucket bucket = client.fetchBucket(BUCKET_NAME).execute();
        bucket.store(user).execute();
    }

    public User get(String UserName) throws RiakRetryFailedException {
        Bucket bucket = client.fetchBucket(BUCKET_NAME).execute();
        return bucket.fetch(UserName, User.class).execute();
    }
}
